### PR TITLE
refactor openapi generation

### PR DESCRIPTION
**Pull request by swaroopar**
refactored openapi generation to download/copy the jar file only at boot time.

### CODE REVIEW

File: modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiGeneratorJarManage.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
+        File existingJarFile = getCliFile();
+        if (existingJarFile.exists()) {
+            log.info(OPENAPI_GENERATOR_FILE_NAME + " already exists.");
+            return;
+        }
+        ClassPathResource resource = new ClassPathResource(OPENAPI_GENERATOR_FILE_NAME);
+        if (resource.exists()) {
+            log.info("openapi-generator-cli.jar found in resources. Copying it.");
```
Comment (jinyangyang222) [inline modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiGeneratorJarManage.java:65]: should be: OPENAPI_GENERATOR_FILE_NAME + " found in resources. Copying it."

File: modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUrlManage.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ */
+@Slf4j
+@Component
+public class OpenApiUrlManage {
+
+    private final String openapiPath;
+    private final Integer port;
+
+    /**
+     * OpenApiUtil constructor.
```
Comment (jinyangyang222) [inline modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUrlManage.java:27]: should be: OpenApiUrlManage constructor.

**Review by jinyangyang222 — COMMENTED** [review]
looks good, Just a few small comments.

File: modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUrlManage.java (hunk 2 above)
Comment (swaroopar) [reply modules/common/src/main/java/org/eclipse/xpanse/common/openapi/OpenApiUrlManage.java:27]: fixed

**Review by iskey — APPROVED** [review]
LGTM

**Pull request merged**